internal/output/disk/disk.go

```diff
@@ -29,33 +31,40 @@ func (d *Disk) TransactionChan() chan<- types.Transaction {
 
 //Init func Initialize the disk output plugin
 func (d *Disk) Init(config types.Config, logger zap.Logger) error {
-	d.FileName = config["filename"].(string)
-	d.FilePath = config["filepath"].(string)
+	if d.FilePath,d.TypeCheck = config["filepath"].(string); !d.TypeCheck {
+		return errors.New("FilePath must be a string")
+	}
 	path.Clean(d.FilePath)
-	d.FilePermission = config["permission"].(os.FileMode)
-	d.Transactions = make(chan types.Transaction, 1)
+	if d.FilePermission,d.TypeCheck = config["permission"].(string); !d.TypeCheck{
+		return errors.New("FilePermission must be from a string")
+	}
+	d.Transactions = make(chan types.Transaction)
 	d.stopChan = make(chan struct{})
 	d.logger = logger
 	return nil
 }
 
-//WriteOnDisk func takes the disk and the transaction
+//WriteOnDisk func takes the transaction
 //that to be written on the disk
-func WriteOnDisk(d *Disk, transaction types.Transaction) {
+func (d *Disk) WriteOnDisk(transaction types.Transaction) {
 	defer d.wg.Done()
-	var err error
 	ack := true
-	d.logger.Info("RECEIVED OUTPUT TRANSACTION...")
-	bytes, _ := ioutil.ReadAll(transaction)
-	if _, err = os.Stat(d.FilePath); os.IsNotExist(err) {
-		err = os.MkdirAll(d.FilePath, os.ModePerm)
+	bytes, err := ioutil.ReadAll(transaction)
+	if err==nil {
+		dir:=filepath.Dir(d.FilePath)
+		if _, err = os.Stat(dir); os.IsNotExist(err) {
+			err = os.MkdirAll(dir, os.ModePerm)
+		}
 	}
 	if err == nil {
-		FilePath := filepath.Join(d.FilePath, d.FileName)
-		err = ioutil.WriteFile(FilePath, bytes, d.FilePermission)
-		d.logger.Info("OUTPUT SUCCESSFUL, Sending Response. ")
-	} else {
-		d.logger.Info("Error in output: ", zap.Error(err))
+		perm32,errParse:=strconv.ParseUint(d.FilePermission,0,32)
```

Comment:
Since these are static, no need to parse and cast at every write. 
you can cache them by doing this on `Init()`